services/core-web/src/components/Forms/ExplosivesPermit/MagazineFormNew.tsx

```diff
@@ -233,11 +216,16 @@ export const MagazineForm = (props) => {
         <Row gutter={48}>
           <Col md={24}>
             {fields.map((field, index) => (
-              <Collapse key={index} className="magazine-collapse margin-large--bottom">
+              <Collapse
+                defaultActiveKey={activeKeys}
+                key={index}
```

Comment:
I am surprised that it let you use the index as the key without an eslint disable. Nothing wonky happens when you try to delete magazines[0] and magazines[1] is expanded or anything?

Reply:
That's actually how it always was.  The problem is, the array being passed is created by redux-form fieldArray.  There is supposed to be a unique id passed by that, but trying it it didn't exist. 😠 

That did make me notice that adding the onChange caused the delete button to be unclickable.  Just added a stopPropagation to fix that.

Reply:
But yeah, all the functions do work 😕

Reply:
That's wacky, I'm used to seeing // eslint-disable-next-line react/no-array-index-key where there's key={index} (why in some places but not in others? I was assuming they were all from --no-verify pushes)

Good that it works, I guess?

Reply:
TS doesn't complain about it here.  Not sure why.  Agreed that it's certainly not a pattern I'd prefer, but I struggled to find a decent alternative.

I actually did find unique id property for the fields, but whenever it triggered this re-render it would change that value as well.